extensions/binding/org.eclipse.smarthome.binding.mqttbroker/src/main/java/org/eclipse/smarthome/binding/mqttbroker/internal/NetworkUtils.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.mqttbroker.internal;
+
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.net.util.SubnetUtils;
+
+/**
+ * Network utility functions for determining all interfaces and assigned IP addresses.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class NetworkUtils {
+    /**
+     * Use this within the class to internally call NetworkInterface.getNetworkInterfaces().
+     * This is done for testing purposes.
+     *
+     * @return
+     */
+    public Enumeration<NetworkInterface> getNetworkInterfaces() {
+        try {
+            return NetworkInterface.getNetworkInterfaces();
+        } catch (SocketException ignored) {
+            // If we are not allowed to enumerate, we return an empty result set.
+            return Collections.enumeration(new ArrayList<NetworkInterface>());
+        }
+
+    }
+
+    /**
+     * Gets every IPv4 Address on each Interface except the loopback
+     * The Address format is ip/subnet
+     *
+     * @return The collected IPv4 Addresses
+     */
+    public Set<String> getInterfaceIPs() {
+        Set<String> interfaceIPs = new HashSet<String>();
+
+        // For each interface ...
+        for (Enumeration<NetworkInterface> en = getNetworkInterfaces(); en.hasMoreElements();) {
+            NetworkInterface networkInterface = en.nextElement();
+            boolean isLoopback = true;
+            try {
+                isLoopback = networkInterface.isLoopback();
+            } catch (SocketException ignored) {
+            }
+            if (!isLoopback) {
+                // .. and for each address ...
+                for (Iterator<InterfaceAddress> it = networkInterface.getInterfaceAddresses().iterator(); it
+                        .hasNext();) {
+
+                    // ... get IP and Subnet
+                    InterfaceAddress interfaceAddress = it.next();
+                    interfaceIPs.add(interfaceAddress.getAddress().getHostAddress() + "/"
+                            + interfaceAddress.getNetworkPrefixLength());
+                }
+            }
+        }
+
+        return interfaceIPs;
+    }
+
+    /**
+     * Get a set of all interface names.
+     *
+     * @return Set of interface names
+     */
+    public Set<String> getInterfaceNames() {
```

Comment:
unused, should be removed (or, if used outside of ESH should be moved to core network utils).